lib/_debug_agent.js

```diff
@@ -117,39 +117,41 @@ Client.prototype.destroy = function destroy(msg) {
 Client.prototype._transform = function _transform(data, enc, cb) {
   cb();
 
-  this.buffer += data;
+  this.buffer = Buffer.concat([this.buffer, data]);
 
   while (true) {
     if (this.state === 'headers') {
       // Not enough data
       if (!this.buffer.includes('\r\n'))
         break;
 
-      if (this.buffer.startsWith('\r\n')) {
+      var bufString = this.buffer.toString('utf8');
+      if (bufString.startsWith('\r\n')) {
         this.buffer = this.buffer.slice(2);
         this.state = 'body';
         continue;
       }
 
       // Match:
       //   Header-name: header-value\r\n
-      var match = this.buffer.match(/^([^:\s\r\n]+)\s*:\s*([^\s\r\n]+)\r\n/);
+      var match = bufString.match(/^([^:\s\r\n]+)\s*:\s*([^\s\r\n]+)\r\n/);
       if (!match)
         return this.destroy('Expected header, but failed to parse it');
 
       this.headers[match[1].toLowerCase()] = match[2];
 
-      this.buffer = this.buffer.slice(match[0].length);
+      this.buffer = this.buffer.slice(Buffer.byteLength(match[0], 'utf8'));
```

Comment:
`this.buffer` could be turned into a `string_decoder.StringDecoder`.  It's more structured than manual buffer->string manipulation and should be more efficient too.